File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {

        string ans = "";

        reverse(s.begin(), s.end());

        for (int i = 0; i < s.size(); i++)
        {
            string words = "";

            while (i < s.size() && s[i] != ' ')
            {
                words += s[i];
                i++;
            }

            reverse(words.begin(), words.end());

            if (words.size() > 0)
            {
                ans += ' ' + words;
            }
        }

        return ans.substr(1);   // remove leading space
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
# class Solution:
#     def reverseWords(self, s: str) -> str:
#         words = s.split()
#         words.reverse()
#         return " ".join(words)

class Solution:
    def reverseWords(self, s: str) -> str:
        words = []
        word = ""

        # Extract words manually
        for ch in s:
            if ch != ' ':
                word += ch
            else:
                if word != "":
                    words.append(word)
                    word = ""

        # Add last word
        if word != "":
            words.append(word)

        # Build answer in reverse order
        answer = ""

        for i in range(len(words) - 1, -1, -1):
            answer += words[i]

            if i != 0:
                answer += " "

        return answer